File: algos/maskify/getEven/geteven.py

def get_even_numbers(arr):
    enum = []
    for n in arr:
        if n % 2 == 0:
            enum.append(n)
    return enum


print(get_even_numbers([2, 4, 5, 6]))


#other solutions#

# def get_even_numbers(arr):
#     return [x for x in arr if x % 2 == 0]


# def is_even(x): return x % 2 == 0


# def get_even_numbers(arr):
#     return list(filter(is_even, arr))


# def get_even_numbers(arr):
#     return [v for v in arr if not v & 1]


# def get_even_numbers(arr):
#     return list(filter(lambda x: not x % 2, arr))


# def get_even_numbers(arr):
#     return [n for n in arr if n % 2 == 0]


# def get_even_numbers(arr):
#     return [a for a in arr if a % 2 == 0]


# def get_even_numbers(arr):
#     return [i for i in arr if i % 2 == 0]


# def get_even_numbers(_): return list(filter(lambda __: not __ % 2, _))


# def get_even_numbers(arr): return [i for i in arr if i % 2 == 0]


# def get_even_numbers(arr):
#     return [num for num in arr if not num % 2]


# def get_even_numbers(arr):
#     return [item for item in arr if not item % 2]


# def get_even_numbers(lst):
#     print(lst)
#     return [x for x in lst if not x % 2]


# def get_even_numbers(arr):
#     return [number for number in arr if number % 2 == 0]


# def get_even_numbers(arr):
#     return [int(num) for num in arr if num % 2 == 0]

File: algos/maskify/getEven/geteven.js
//n is for numbers in array
//return n as a key
//need to modulate the nums to get results
//filter each num for result and display all evens by using modulas opperation, 

function getEvenNumbers(numbersArray) {
    return numbersArray.filter((n => { return n % 2 == 0; }))
}
  // filter out the odd numbers


//other solutions//

/* function getEvenNumbers(numbersArray){
  return numbersArray.filter(function(num){return !(num % 2) })  // 0 is falsy
}

function getEvenNumbers(numbersArray){
  return numbersArray.filter(function(n){
    return n % 2 == 0;
  });
}


const getEvenNumbers = numbersArray => numbersArray.filter(num => num % 2 == 0);


const getEvenNumbers = arr =>
  arr.filter(x => !(x & 1));
  

  function isEven(num){
  return num%2==0
}

function getEvenNumbers(numbersArray){
  // filter out the odd numbers
  var filtered = numbersArray.filter(isEven);
  return filtered;
}



function getEvenNumbers(numbersArray){
  // filter out the odd numbers
  var evenNums = numbersArray.filter(function(num) {
    return (num % 2) === 0;
  });

  return evenNums;
}



const getEvenNumbers = numbersArray => numbersArray.filter(a => a % 2 === 0);





function getEvenNumbers(numbersArray){
  // filter out the odd numbers
  return numbersArray.filter(function(l){return l%2 == 0});
}



const getEvenNumbers = numbersArray =>
  numbersArray.filter(val => !(val&1));





function checkOdd(number) {
  if (number % 2 == 0)
    return number;
}

function getEvenNumbers(numbersArray){
  // filter out the odd numbers
  const result = numbersArray.filter(checkOdd);

  console.log(result);

  return result;

}




  */